src/index.js

```diff
@@ -1,39 +1,178 @@
+const completeItem = () => {
+  const localData = localStorage.getItem('items');
+  const parsedData = JSON.parse(localData);
+  const eachItem = document.querySelectorAll('.eachItem');
+  for (let i = 0; i < eachItem.length; i += 1) {
+    if (eachItem[i].classList.contains('strike')) {
+      parsedData[i].completed = true;
+    } else {
+      parsedData[i].completed = false;
+    }
+    localStorage.setItem('items', JSON.stringify(parsedData));
+  }
+};
+
+// Edit Tasks
+const editItems = (oldItem) => {
+  const newInput = document.createElement('input');
+  newInput.type = 'text';
+  newInput.classList.add('new-input');
+  newInput.value = oldItem.textContent;
+  oldItem.replaceWith(newInput);
+  const local = localStorage.getItem('items');
+  const data = JSON.parse(local);
+  newInput.addEventListener('keypress', (e) => {
+    const newId = newInput.parentElement.id;
+    if (e.key === 'Enter') {
+      const oldItem = document.createElement('div');
+      oldItem.textContent = newInput.value;
+      newInput.replaceWith(oldItem);
+      const editedContainers = document.querySelectorAll('.new-task');
+      for (let i = 0; i < editedContainers.length; i += 1) {
+        if ((editedContainers[i].id) === newId) {
+          data[i].description = newInput.value;
+          localStorage.setItem('items', JSON.stringify(data));
+        }
+      }
+    }
+  });
+};
+
+// Remove Tasks
+const removeItems = (li) => {
+  listOfTasks.removeChild(li);
+  let count = 1;
+  const parsedItems = localStorage.getItem('items');
+  let localData = JSON.parse(parsedItems);
+  // Filter tasks that are true
+  taskArray = JSON.parse(localStorage.getItem('items'));
+  taskArray.splice((li.id) - 1, 1);
+  localData = taskArray;
+  // Update index of tasks
+  localData.map((item) => {
+    item.index = count;
+    count += 1;
+    return null;
+  });
+  localStorage.setItem('items', JSON.stringify(localData));
+};
+
+// Add new task
+const addNewTask = (newDescription) => {
+  const newTask = document.createElement('div');
+  newTask.classList.add('list');
+  newTask.innerHTML += `
+<input type="checkbox" class="checkbox">
+<span>${newDescription}</span>
+<span class="ellipsis">
+<i class="fa-solid fa-ellipsis-vertical"></i>
+</span>
+<span class="trashcan">
+<i class="fa-solid fa-trash-can"></i>
+</span>
+`;
+  listOfTasks.appendChild(newTask);
+  const checkbox = document.querySelectorAll('.checkbox');
+  checkbox.forEach((checkboxInput) => {
+    checkboxInput.addEventListener('click', () => {
+      checkboxInput.nextElementSibling.classList.toggle('strike');
+      checkboxInput.parentNode.classList.toggle('clicked-on');
+      const ellipsisIcon = checkboxInput.parentNode.childNodes[5];
+      ellipsisIcon.classList.toggle('inactive-ellipsis');
+      const trashIcon = checkboxInput.parentNode.childNodes[7].childNodes[1];
+      trashIcon.classList.toggle('active-trash');
+      completeItem();
+    });
+  });
```

Comment:
- [ ] Greate job implementing the required functionality however they need to be on a new JavaScript file, kindly move these functions to a new file so your code is divided and cleaner